### PR TITLE
Add: liked, history, watchlater feature

### PR DESCRIPTION
**Pull request by gauravsinhaweb**
files to check:

- likedVideos.jsx
- watchLater.jsx
- history.jsx

### CODE REVIEW

File: src/components/cards/ThumbnailCard.jsx [hunk 1]
```diff
@@ -1,13 +1,35 @@
 import React from "react";
+import { AiOutlineClockCircle } from "react-icons/ai";
+import { useNavigate } from "react-router-dom";
+import {
+  useDataContext,
+  useFeatureContext,
+} from "../../context/useContext-index";
+import { useAxios } from "../../services/auth/useAxios";
```
Comment (Ankur9669) [inline src/components/cards/ThumbnailCard.jsx:9]: Suggestion: You can import all files in one single file

File: src/components/cards/ThumbnailCard.jsx [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
+      videoData,
+      featureDispatch,
+      "WATCH_LATER"
+    );
   return (
     <>
       <div className="w-full flex flex-col cursor-pointer">
-        <div className="relative">
+        <div onClick={() => navigate(`/video/${id}`)} className="relative">
           <img
```
Comment (Ankur9669) [inline src/components/cards/ThumbnailCard.jsx:33]: fix: alt tag is missing

File: src/components/navbar/Navbar.jsx [hunk 3]
```diff
@@ -2,21 +2,25 @@ import React from "react";
...
 import { GiHamburgerMenu } from "react-icons/gi";
 import { HiOutlineLogout } from "react-icons/hi";
 import { MdExplore, MdPlaylistPlay } from "react-icons/md";
 import { Link, useNavigate } from "react-router-dom";
 
-export const Navbar = () => {
+export const Navbar = (props) => {
+  const { setSearchValue } = props;
   const navigate = useNavigate();
   const logoutHandler = () => {
```
Comment (Ankur9669) [inline src/components/navbar/Navbar.jsx:14]: Fix: In the case of logout you also need to remove the data of the user that you have saved in the context.

File: src/pages/hero/Hero.jsx [hunk 4]
```diff
@@ -1,15 +1,20 @@
 import React from "react";
 import { useNavigate } from "react-router-dom";
 import { CategoryCard, ThumbnailCard } from "../../components/index-component";
-import { useDataContext } from "../../context/data-context";
-import { categoryFilter } from "../../utils/util-index";
+import {} from "../../context/filter-context";
+import { useDataContext } from "../../context/useContext-index";
+import { categoryFilter, searchFilter } from "../../utils/util-index";
+
```
Comment (Ankur9669) [inline src/pages/hero/Hero.jsx:7]: Can import all files in one single file

File: src/pages/hero/Hero.jsx [hunk 5]
```diff
@@ -1,15 +1,20 @@
 import React from "react";
 import { useNavigate } from "react-router-dom";
 import { CategoryCard, ThumbnailCard } from "../../components/index-component";
-import { useDataContext } from "../../context/data-context";
-import { categoryFilter } from "../../utils/util-index";
+import {} from "../../context/filter-context";
```
Comment (Ankur9669) [inline src/pages/hero/Hero.jsx:4]: Is this intentional??

File: src/pages/videoDetail/VideoDetail.jsx [hunk 6]
```diff
@@ -1,19 +1,50 @@
...
+import { MdPlaylistAdd } from "react-icons/md";
+import { useParams } from "react-router-dom";
 import { ThumbnailCard } from "../../components/index-component";
-import { useDataContext } from "../../context/data-context";
+import {
+  useDataContext,
+  useFeatureContext,
+} from "../../context/useContext-index";
+import { useAxios } from "../../services/auth/useAxios";
+import { dislikeHandler } from "../../utils/util-index";
```
Comment (Ankur9669) [inline src/pages/videoDetail/VideoDetail.jsx:13]: Same import all the files in one single file

File: src/utils/feature-function/dislikeHandler.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import axios from "axios";
+
+export const dislikeHandler = async (e, id, dispatch, likedVideos) => {
+  e.preventDefault();
+  try {
+    const res = await axios.delete(`/api/user/likes/${id}`);
+    if (res.status === 200) {
+      console.log(res);
```
Comment (Ankur9669) [inline src/utils/feature-function/dislikeHandler.js:8]: Suggestion: Remove console.log()

**Review by Ankur9669 — CHANGES_REQUESTED** [review]
The code looks good to me, although i was not able to use your app because functionalities are not working.

**Review by Ankur9669 — APPROVED** [review]
Edit: The files are working for now, you can merge it
Suggestion: Use toasts for indication, because the user will not know whether the video is liked or disliked or it has gone in watch later or not

**Review by Sahil25061999 — APPROVED** [review]
code looks good , great ui, just a ui suggestion add some kind of effect once the video is added to watch later or is liked.

File: src/components/cards/ThumbnailCard.jsx (hunk 2 above)
Comment (gauravsinhaweb) [reply src/components/cards/ThumbnailCard.jsx:33]: alt tag is already there!

File: src/components/navbar/Navbar.jsx (hunk 3 above)
Comment (gauravsinhaweb) [reply src/components/navbar/Navbar.jsx:14]: ok i'll do that

File: src/pages/hero/Hero.jsx (hunk 5 above)
Comment (gauravsinhaweb) [reply src/pages/hero/Hero.jsx:4]: nope, updated

**Pull request merged**